pages/api/setup/[token]/directory-sync/index.ts

```diff
@@ -0,0 +1,65 @@
+import type { NextApiRequest, NextApiResponse } from 'next';
+import type { DirectoryType } from '@lib/jackson';
+import jackson from '@lib/jackson';
+
+export const handler = async (req: NextApiRequest, res: NextApiResponse) => {
+  const { method } = req;
+  const { setupLinkController } = await jackson();
+  const token = req.query.token;
+  let data, error;
+  if (!token) {
```

Comment:
I don't think you have to do this checking on API endpoints, method `getByToken` is already doing it. Am i right? This way you can just return the `data or error` returned by `getByToken`. 

https://github.com/boxyhq/jackson/blob/3e5d83ccbdba29a0c266ff44c0934244cf8a631d/pages/api/setup/%5Btoken%5D/directory-sync/index.ts#L10-L34

At the moment, this piece of code is duplicated in many places.